File: 05_arrays/05_07_move_zeroes/move_zeroes.js
const moveZeroes = (arr = []) => {
    let j = 0;
    for (let index = 0; index < arr.length; index++) {
        if (arr[index] != 0) {
            [arr[index], arr[j]] = [arr[j], arr[index]];
            j++;
        }
    }
}

const array = [0, 1, 0, 3, 12]
console.log(array);
moveZeroes(array);
console.log(array);

File: 05_arrays/05_07_move_zeroes/move_zeroes.cpp
#include <bits/stdc++.h>

using namespace std;

void moveZeroes(vector<int>& nums) {
    int n = nums.size();
    int j = 0;
    for (size_t i = 0; i < n; i++)
    {
        if (nums[i] != 0)
        {
            swap(nums[i], nums[j++]);
        }
    }
}

int main() {
    vector<int> nums = {0, 1, 0, 3, 12};
    int n = nums.size();

    cout << "Elements:- " << endl;
    for (size_t i = 0; i < n; i++)
    {
        cout << nums[i] << " ";
    }
    cout << endl;

    moveZeroes(nums);

    cout << "Zeroes moved:- " << endl;
    for (size_t i = 0; i < n; i++)
    {
        cout << nums[i] << " ";
    }
    cout << endl;
}